File: LeetCode Solutions/Problems/870. Advantage Shuffle.cpp
class Solution {
public:
    vector<int> advantageCount(vector<int>& nums1, vector<int>& nums2) {
        vector<int> order = vector<int>(nums2.size()), permutation = vector<int>(nums1.size());
        for(int i=0; i<nums2.size(); i++){
            order[i] = i;
        }
        sort(order.begin(), order.end(), [&](int a, int b) {return nums2[a] > nums2[b];});
        sort(nums1.begin(), nums1.end(), greater<>());
        int i = 0, j = nums2.size()-1;
        for(int ix : order){
            permutation[ix] = nums1[i] > nums2[ix] ? nums1[i++] : nums1[j--];
        }
        return permutation;
    }
};

File: LeetCode Solutions/Problems/870. Advantage Shuffle.java
class Solution {
    public int[] advantageCount(int[] nums1, int[] nums2) {
        Integer[] order = new Integer[nums2.length];
        int[] permutation = new int[nums1.length];
        for(int i=0; i<nums2.length; i++){
            order[i] = i;
        }
        Arrays.sort(order, (a,b) -> Integer.compare(nums2[b], nums2[a]));
        Arrays.sort(nums1);
        int i = 0, j = nums2.length-1;
        for(int ix : order){
            permutation[ix] = nums1[j] > nums2[ix] ? nums1[j--] : nums1[i++];
        }
        return permutation;
    }
}